okio/src/main/java/okio/Buffer.java

```diff
@@ -775,6 +840,93 @@ public void clear() {
     return writeLong(reverseBytesLong(v));
   }
 
+  @Override public Buffer writeDecimalLong(long v) {
+    if (v == 0) {
+      // Both a shortcut and required since the following code can't handle zero.
+      return writeByte('0');
+    }
+
+    boolean negative = false;
+    if (v < 0) {
+      v = -v;
+      if (v < 0) { // Only true for Long.MIN_VALUE.
+        return writeUtf8("-9223372036854775808");
+      }
+      negative = true;
+    }
+
+    // Binary search for character width which favors matching lower numbers.
+    int width;
+    if (v < 100000000L) {
+      if (v < 10000L) {
+        if (v < 100L) {
+          width = v < 10L ? 1 : 2;
+        } else {
+          width = v < 1000L ? 3 : 4;
+        }
+      } else {
+        if (v < 1000000L) {
+          width = v < 100000L ? 5 : 6;
+        } else {
+          width = v < 10000000L ? 7 : 8;
+        }
+      }
+    } else {
+      if (v < 1000000000000L) {
+        if (v < 10000000000L) {
+          width = v < 1000000000L ? 9 : 10;
+        } else {
+          width = v < 100000000000L ? 11 : 12;
+        }
+      } else {
+        if (v < 1000000000000000L) {
+          if (v < 10000000000000L) {
+            width = 13;
+          } else {
+            width = v < 100000000000000L ? 14 : 15;
+          }
+        } else {
+          if (v < 100000000000000000L) {
+            width = v < 10000000000000000L ? 16 : 17;
+          } else {
+            width = v < 1000000000000000000L ? 18 : 19;
+          }
+        }
+      }
+    }
+    if (negative) {
+      ++width;
+    }
+
+    Segment tail = writableSegment(width);
+    byte[] data = tail.data;
+    int pos = tail.limit + width; // We write backwards from right to left.
+    while (v != 0) {
+      int digit = (int) (v % 10);
+      data[--pos] = DIGITS[digit];
+      v /= 10;
+    }
+    if (negative) {
+      data[--pos] = '-';
+    }
+
+    tail.limit += width;
+    size += width;
+    return this;
+  }
+
+  @Override public Buffer writeHexadecimalUnsignedLong(long v) {
```

Comment:
Writing the leading `0` chars makes this lame for HTTP chunked encoding. See `HttpConnection.writeHex()`